running-locally.md

```diff
@@ -0,0 +1,38 @@
+doing our best to duplicate the [parameters](https://github.com/scala-steward-org/scala-steward/blob/main/docs/help.md) which are [setup by the Scala Steward GitHub Action](https://github.com/scala-steward-org/scala-steward-action/blob/31bf30f6a29b33723e5b6d89a9a8f4cf23b05924/src/action/main.ts#L49-L71)
+when we run it in [our GitHub workflow](https://github.com/guardian/scala-steward-public-repos/blob/2c3ae97fe168c0887036fac81752b7d406653823/.github/workflows/public-repos-scala-steward.yml).
+
+1. Clone Scala Steward's codebase: [scala-steward-org/scala-steward](https://github.com/scala-steward-org/scala-steward).
+   We recommend [these alterations](https://github.com/scala-steward-org/scala-steward/compare/main...rtyley:scala-steward:temporary-tweaks-for-debugging) to the code:
+   * Uncomment the [`run / javaOptions`](https://github.com/scala-steward-org/scala-steward/blob/609104a04cd536418ebcda553e7f48177a5264ae/build.sbt#L213)
+     build.sbt line to enable debugging.
+   * Disable actual creation/update of PRs by removing [the call to `NurtureAlg.nurture()`](https://github.com/scala-steward-org/scala-steward/blob/42da4fad649dfc5e92147b3e8165a1bf860b2629/modules/core/src/main/scala/org/scalasteward/core/application/StewardAlg.scala#L82)
+   * Optionally, filter the repos list to focus on a single repo that you find interesting (eg [`amigo`](https://github.com/guardian/amigo))
+2. Clone _this_ repo, [guardian/scala-steward-public-repos](https://github.com/guardian/scala-steward-public-repos),
+   for our [scala-steward.conf](https://github.com/guardian/scala-steward-public-repos/blob/main/scala-steward.conf)
+   and [REPOSITORIES.md](https://github.com/guardian/scala-steward-public-repos/blob/main/REPOSITORIES.md) files.
+3. Create a workspace folder for Scala Steward to work in, eg. `~/code/scala-steward-workspace`
+4. Create a blank `askpass.sh` file (normally this would need to
+   [contain a valid user GitHub token](https://github.com/scala-steward-org/scala-steward-action/blob/190e7f36cd057dfb5d1b6fde688e6399ef249560/src/modules/workspace.ts#L114),
+   but it appears that's only necessary if the code is actually going to push commits to GitHub).
+   ```shell
+   touch ~/code/scala-steward-workspace/askpass.sh
+   chmod 755 ~/code/scala-steward-workspace/askpass.sh
+   ```
+5. Download a [temporary GitHub App private key](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps)
+   (a `.private-key.pem` file) for our [gu-scala-steward-public-repos](https://github.com/apps/gu-scala-steward-public-repos)
+   GitHub App. Remember to [**DELETE THIS KEY**](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps#deleting-private-keys) when you've finished your debugging work.
```

Comment:
Is this accessible to everyone? Or just org admins?

Reply:
I think only [App Managers](https://docs.github.com/en/apps/maintaining-github-apps/about-github-app-managers) for the specific GitHub App can generate/download new keys for the app.

Checking [App managers for our gu-scala-steward-public-repos app](https://github.com/organizations/guardian/settings/permissions/integrations/gu-scala-steward-public-repos/managers), currently the only app managers are `Organization owner`s, but there _is_ an option at the bottom to grant more users App Manager access, if we definitely wanted to.

Reply:
Clarified this in the doc with 341732a805b68354cb735880e686b2d8faf2e665.